### ISSUE TITLE
Distributed cache: loading and coordination

### ISSUE THREAD

**Issue by robinbraemer** [ory/keto#312]
**Is your feature request related to a problem? Please describe.**

Caching is to be considered later and can be added behind the Persister interface.
Groupcache (by Google) is a library for coordinating distributed caching and cache loading. 
https://github.com/mailgun/groupcache

**Comment by robinbraemer** [comment]
In order to form and resiliently maintain a cluster of members aware of its peers we can use:
- https://github.com/hashicorp/memberlist 
  - sample project: https://github.com/asim/memberlist

**Comment by aeneasr** [comment]
Implementing a chatty, self-healing, globally distributed groupcache is extremely hard. I would suggest we use some other project like etcd or memcached to back this instead.

**Comment by robinbraemer** [comment]
Another such system would.need to be eventually consistent and provide coordinated cache-loading in case of ache miss.

**Comment by aeneasr** [comment]
For now, we could use a non-shared cache such as https://github.com/dgraph-io/ristretto to reduce load times and reduce pressure at least for some of the requests.

**Comment by robinbraemer** [comment]
Yes, ristretto would be the best single instance local cache I could think of to fallback to.
